picky-asn1-x509/src/private_key_info.rs

```diff
@@ -328,9 +360,30 @@ impl RsaPrivateKey {
     }
 }
 
+/// [Elliptic Curve Private Key Structure] (https://datatracker.ietf.org/doc/html/rfc5915#section-3)
+///
+/// EC private key information SHALL have ASN.1 type ECPrivateKey:
+///
+/// ```not_rust
+/// ECPrivateKey ::= SEQUENCE {
+///     version        INTEGER { ecPrivkeyVer1(1) } (ecPrivkeyVer1),
+///      privateKey     OCTET STRING,
+///      parameters [0] ECParameters {{ NamedCurve }} OPTIONAL,
+///      publicKey  [1] BIT STRING OPTIONAL
+///    }
+/// ```
+
+#[derive(Serialize, serde::Deserialize, Debug, Clone, PartialEq)]
+pub struct ECPrivateKey {
+    pub private_key: OctetStringAsn1,
+    pub parameters: ExplicitContextTag0<EcParameters>,
+    pub public_key: ExplicitContextTag1<BitStringAsn1>,
+}
```

Comment:
This is not obeying the `ECPrivateKey` format as specified in the spec. If this helps: you should be able to deserialize content of a "EC PRIVATE KEY" pem by deserializing into this struct only.